Assets/_Game/Scripts/combat/HealthComponent.cs

```diff
@@ -54,7 +53,16 @@ private void MaxHealthAdd(HeartContainerObject _container, GameObject _obj)
             }
         }
     }
-
+    public void MaxHealthAdd(int _amountToAdd)
+    {
+        maxHealth += _amountToAdd;
+        health = maxHealth;
+        if(gameObject.tag == "Player")
+        {
+            SaveHealth();

```

Comment:
You don't want to save the health when you up it. It should only be saved at the end of levels or we'll get bugs when people reset. Probably means the SaveHealth call should be removed from the other two methods as well. Apparently I missed that when fixing the bug previously